Database/Repositories/Implementations/TournamentsRepository.cs

```diff
@@ -105,12 +111,53 @@ public async Task<ICollection<Tournament>> GetAsync(
 
         if (ruleset.HasValue)
         {
-            query = query.Where(x => x.Ruleset == ruleset.Value);
+            query = query.Where(t => t.Ruleset == ruleset.Value);
+        }
+
+        if (!string.IsNullOrEmpty(name))
```

Comment:
All of these `if` statements are really ugly, would prefer adding extension methods for all of these and using a builder pattern. Such as:

```cs
query = query.WhereSearchKey(searchKey).WhereMinDate(dateMin)
...
```

The extensions could support nullables.